llvm/lib/Object/MachOUniversalWriter.cpp

```diff
@@ -117,10 +117,16 @@ static Expected<MachoCPUTy> getMachoCPUFromTriple(StringRef TT) {
   return getMachoCPUFromTriple(Triple{TT});
 }
 
+static Expected<MachoCPUTy>
+getMachoCPUFromObjectFile(const MachOObjectFile *O) {
+  return std::make_pair(O->getHeader().cputype, O->getHeader().cpusubtype);
```

Comment:
It looks like Expected might be unnecessary here
nit: since O is not null, maybe just pass const MachObjectFile &O